mirgecom/eos.py

```diff
@@ -0,0 +1,252 @@
+__copyright__ = """
+Copyright (C) 2020 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import numpy as np
+import numpy.linalg as la  # noqa
+from pytools.obj_array import flat_obj_array
+from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
+
+r"""
+This module is designed provide Equation of State
+objects used to compute and manage the relationships
+between and among state and thermodynamic variables.
+"""
+
+
+class GasEOS:
+    r"""Implements an object designed to provide methods
+    for implementing and computing relations between
+    fluid or gas state variables.
```

Comment:
Maybe we should, that is if we plan on having the EOS handle the non-dimensionalization. This needs to be a part of an extended discussion on how non-dimensionalization should be handled.   There is a lot to be said about keeping the solution state dimensionless at all times, while applying dimensionalization at say viz-write time, or just supplying the non-dimensionalization parameters so that the user can apply them at their discretion.